new_session_test.go

```diff
@@ -1314,6 +1314,27 @@ func (s *testSessionSuite) TestDeletePanic(c *C) {
 	tk.MustExec("delete from `t` where `c` = ?", 2)
 }
 
+func (s *testSessionSuite) TestStatementErrorInTransaction(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("drop table if exists test;")
+	tk.MustExec(`create table test (
+		  a int(11) DEFAULT NULL,
+		  b int(11) DEFAULT NULL
+	) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin;`)
+	tk.MustExec("insert into test values (1, 2), (1, 2), (1, 1), (1, 1);")
+	tk.MustExec("commit")
+
+	tk.MustExec("start transaction;")
+	// In the transaction, statement error should not rollback the transaction.
+	_, err := tk.Exec("update tset set b=11 where a=1 and b=2;")
+	c.Assert(err, NotNil)
+	// Test for a bug that last line rollback and exit transaction, this line autocommit.
+	tk.MustExec("update test set b = 11 where a = 1 and b = 2;")
+	tk.MustExec("rollback")
```

Comment:
If it is `commit` here, what will happen? I think it will rollback either.

Reply:
If it's `commit`, it will not rollback, it will commit this line

```
update test set b = 11 where a = 1 and b = 2
```

MySQL ignore the statement error if user doesn't handle it, and continue to commit the remain statements.
We keep the same behavior with MySQL.

Reply:
No statement rollback in TiDB. Try to keep the same behavior with MySQL will cause some bugs.

Reply:
You're right. I'll fix it in another PR.